model/godleyTableWindow.cc

```diff
@@ -483,15 +483,17 @@ namespace minsky
     motionRow=motionCol=-1;
     if (selectedRow==0)
       {
-        if (c>0 && c!=selectedCol)
+        if (c>0 && selectedCol!=0 && c!=selectedCol)      // Cannot move Intitial Conditions cell from col 0 to one with a "NoAssetClass" designation. For ticket 1064/1066
           godleyIcon->table.moveCol(selectedCol,c-selectedCol);
       }
     else if (r>0 && selectedCol==0)
       {
-        if (r!=selectedRow && !godleyIcon->table.initialConditionRow(selectedRow) && !godleyIcon->table.initialConditionRow(r))  // Cannot move Intitial Conditions row. For ticket 1064
+		// Cannot move Intitial Conditions row. For ticket 1064. Also cannot move last row down, i.e., button widgets have to be used to create new rows, otherwise an error occurs. For ticket 1066.  
+        if (r!=selectedRow && r<int(godleyIcon->table.rows()) && !godleyIcon->table.initialConditionRow(selectedRow) && !godleyIcon->table.initialConditionRow(r))  
           godleyIcon->table.moveRow(selectedRow,r-selectedRow);
       }
-    else if ((c!=selectedCol || r!=selectedRow) && c>0 && r>0 && (c!=0 || r!=1) && (selectedCol!=0 || selectedRow!=1)) // Cannot swap cell(1,0) with another. For ticket 1064
+     // Cannot swap cell(1,0) with another. For ticket 1064. Also cannot move cells outside an existing Godley table to create new rows or columns. For ticket 1066.  
+    else if ((c!=selectedCol || r!=selectedRow) && (selectedCol!=0 || selectedRow!=1) && c>0 && r>0 && selectedRow<int(godleyIcon->table.rows()) && r<int(godleyIcon->table.rows()) && c <int(godleyIcon->table.cols()) && selectedCol<int(godleyIcon->table.cols()))
```

Comment:
What happened to (c!=0 || r!=1)?

Reply:
Watch out for the else clause on line 502. That might be triggered now with the initial conditions cell, or out of bounds cell reference. I think it might be best, and cleaner to separate out the no op cases into an early return if statement at the top of the function block:

if (c==0 && r==1 || selectCol==0 && selectedRow==1 || c>=cols() || selectedCol>=cols() || r>=rows() || selectedRow>=rows()) return;

Reply:
We don't need to redraw of there is no change of state.

Reply:
Yikes, should not have removed (c!=0 || r!=1). It fell by the wayside during my attempts to shorten the if-clause.

Moving the no op cases to the top is a great idea, thanks. Will implement it right away.